### PR TITLE
sc_released cleanup

### PR DESCRIPTION
**Pull request by mktnk3**
This aligned aspec `sc_released` with Haskell and C definitions, by removing the conjunct on sufficiency.

### CODE REVIEW

File: proof/invariant-abstract/DetSchedSchedule_AI.thy [hunk 1]
```diff
@@ -3304,6 +3303,11 @@ lemma possible_switch_to_valid_sched_strong:
   apply (wpsimp wp: set_scheduler_action_swt_valid_sched'
                     tcb_sched_enqueue_valid_sched reschedule_valid_sched_except_blocked_const
                     reschedule_required_valid_blocked thread_get_wp')
+  apply (prop_tac "pred_map (\<lambda>a. \<exists>y. a = Some y) (tcb_scps_of s) target")
```
Comment (michaelmcinerney) [inline proof/invariant-abstract/DetSchedSchedule_AI.thy:3306]: You could use the `bound` abbreviation here, if you'd like

File: proof/invariant-abstract/DetSchedSchedule_AI.thy [hunk 2]
```diff
@@ -5178,10 +5182,17 @@ lemma sched_context_resume_valid_sched_except_blocked:
    sched_context_resume scp
    \<lbrace>\<lambda>_. valid_sched_except_blocked\<rbrace>"
   apply (wpsimp wp: postpone_valid_sched_except_blocked is_schedulable_wp' get_tcb_queue_wp thread_get_wp'
-              simp: sched_context_resume_def refill_sufficient_def get_refills_def refill_ready_def)
-  apply (clarsimp simp: valid_sched_def sc_at_pred_n_def get_tcb_ko_at obj_at_def)
-  apply (frule (1) heap_refs_retract_atD, clarsimp simp: vs_all_heap_simps)
-  apply (prop_tac "\<not> released_sc_tcb_at tp s", clarsimp simp: vs_all_heap_simps refill_ready_def refill_sufficient_def)
+              simp: sched_context_resume_def get_refills_def)
+  apply (clarsimp simp: obj_at_def)
+  apply (prop_tac "heap_ref_eq y scp (sc_tcbs_of s)")
```
Comment (michaelmcinerney) [inline proof/invariant-abstract/DetSchedSchedule_AI.thy:5187]: Is it at all nicer to `rename_tac` this `y`? I can see there is a `tp` further down, so maybe this is a bit of a confusing one

File: proof/refine/ARM/TcbAcc_R.thy [hunk 3]
```diff
@@ -2451,7 +2453,12 @@ lemma rescheduleRequired_corres_weak:
     apply simp
    apply (wpsimp wp: get_sched_context_no_fail simp: get_sc_refill_ready_def)
    apply (clarsimp intro!: no_ofailD[OF no_ofail_read_sc_refill_ready] simp: obj_at_def is_sc_obj)
-
+  apply (rule_tac F="sufficient = True" in corres_req)
```
Comment (michaelmcinerney) [inline proof/refine/ARM/TcbAcc_R.thy:2456]: You can just write "sufficient" here

File: proof/refine/ARM/Tcb_R.thy [hunk 4]
```diff
@@ -2981,8 +2986,7 @@ lemma get_sc_released_corres:
    apply (wpsimp wp: no_ofail_gets_the readRefillReady_no_ofail)
   apply (clarsimp simp: sc_released_def readRefillReady_def readSchedContext_def
                  dest!: readObject_misc_ko_at')
-  apply normalise_obj_at'
-  apply (subgoal_tac "sc_active sc = (0 < scRefillMax v')")
+  apply (subgoal_tac "sc_active sc = (0 < scRefillMax sca)")
```
Comment (michaelmcinerney) [inline proof/refine/ARM/Tcb_R.thy:2989]: Maybe not worth it, but if may be nice to `rename_tac` this `sca` too, especially considering it's changed its generated name

**Review by michaelmcinerney — APPROVED** [review]
Cool

**Review by corlewis — APPROVED** [review]

**Pull request merged**